extension/micro-core/src/main/java/cloud/piranha/micro/MicroInnerDeployer.java

```diff
@@ -185,8 +185,8 @@
             
             webApplicationServer.initialize();
             webApplicationServer.start();
-            
-            httpServer = new DefaultHttpServer(port, webApplicationServer, false);
```

Comment:
Why is false removed here? We need to be able to support running in SSL mode. I am not sure if we ported that from Piranha Server to Piranha Micro yet ...

Reply:
Note if we did not completely port it yet then please just pass in false and we can address it at a later date

Reply:
As I'm creating the server through reflection, I'll have to try more than one constructor. If it is OK I can add right now

Reply:
OK lets rework the implementations to have only one constructor and then add getters/setters to the HttpServer API and adjust each implementation accordingly. Create everything needed in the default constructor and where that is not possible create it in the start method. I hope that makes sense?

Reply:
Looks good to me

Reply:
The default will be a no-arg constructor or a three arguments constructor (serverPort, processor, ssl) ?